frontend/javascripts/oxalis/view/right-border-tabs/segments_tab/segments_view.tsx

```diff
@@ -406,16 +414,41 @@ class SegmentsView extends React.Component<Props, State> {
         searchableTreeItemList.push(item);
       });
 
-      return {
+      updateStateObject = {
         groupTree: generatedGroupTree,
         searchableTreeItemList,
         prevProps: nextProps,
       };
-    } else {
+    }
+    if (prevState.prevProps?.isosurfaces !== isosurfaces) {
+      //if any segment is invisible, set visibility false, so that the first action is to make all meshes visible
+      const newVisibleMap: { [groupId: number]: boolean } = {};
+      const segmentGroupsWithRoot = [...segmentGroups, rootGroup];
+      segmentGroupsWithRoot.forEach((group) => {
+        let isVisible = true;
+        const segmentsOfGroup = groupToSegmentsMap[group.groupId];
+        if (segmentsOfGroup == null) return;
+        const visibilityOfSegments: number[] = segmentsOfGroup.map((segment) => {
```

Comment:
Could this be simplified to something like this?

```ts
        const isSomeSegmentLoadedAndInvisible = segmentsOfGroup.some((segment) => {
          const segmentIsosurface = isosurfaces[segment.id];
          // Only regard loaded, but invisible meshes
          return segmentIsosurface != null && !isosurfaces[segment.id].isVisible;
        });

        newVisibleMap[group.groupId] = !isSomeSegmentLoadedAndInvisible;
```

(not tested)

Reply:
oh right, this is so much nicer